analysis_options.yaml

```diff
@@ -78,9 +78,12 @@ linter:
 dart_code_metrics:
   rules:
     - prefer-trailing-comma
+  metrics-exclude:
+    - test/**
```

Comment:
why should we exclude tests?

Reply:
It complains about things like cyclomatic complexity, I tried to exclude it for just one file, but @erickzanardo suggested that we'd just exclude all the test files.

Reply:
I would love to prefer-trailing-comma on tests are well, can't we just bump those constants we don't care about?

Reply:
It only excludes the metrics, not the rules, so prefer-trailing-comma will still be active. :)

Reply:
ah great!

Reply:
well maybe we should just do metrics-excludes: *? and remove the custom metrics. we don't use it anyway

Reply:
I disagree, metrics can help us keep our codebase clean.

We could tune it as we understand it better. But I don't see why disable it all the way

Reply:
but you said yourself:

![image](https://user-images.githubusercontent.com/882703/108632428-e5f79580-743c-11eb-8b8f-b9d5ccdb5696.png)

I thought we didn't want this. if we do I think we should reconsider what we have and why. like number of parameters ok I think that is good, LOC per file I am over the fence, cyclomatic complexity I don't think that is a good measure.

Also if we want to keep I think we should not exclude tests. I want our test code to be as pretty as our src code (no matter what definition of pretty we choose).

Reply:
I was just explaining why the numbers of the parameters I have choose.

I think that metrics not necessarily applies to test code, as testing code has a very different way of being written than normal code.

The linter will still be checking the testing code, so it will remain pretty.

I mentioned that we should tune these numbers to what we think it nice, also agree that ciclomatic complexity is not a very good indicator. The only think I am disagreeing is to turn metrics everywhere, we can take advantage of its indicators.